src/subdomains/generic/kyc/services/integration/kyc-document.service.ts

```diff
@@ -54,8 +61,18 @@ export class KycDocumentService {
     name: string,
     data: Buffer,
     contentType: ContentType,
+    isProtected: boolean,
+    kycStep?: KycStep,
     metadata?: Record<string, string>,
   ): Promise<string> {
+    await this.kycFileService.createKycFile({
+      name: name,
+      type: type,
+      protected: isProtected,
+      userData: await this.userDataService.getUserData(userDataId),
+      kycStep,
+    });
```

Comment:
Have `userData` instead of `userDataId` as a parameter (no need to refetch the entity)